content/en/docs/reference/node/kubelet-checkpoint-api.md

```diff
@@ -25,6 +25,87 @@ should create the checkpoint archive to be only accessible by the `root` user. I
 is still important to remember if the checkpoint archive is transferred to another
 system all memory pages will be readable by the owner of the checkpoint archive.
 
+## Security risks and mitigation strategies
+
+1. **Exposure of sensitive data**: When a container is checkpointed, all memory pages,
+   including private data and encryption keys, are saved to the local disk. If the 
+   checkpoint archive is accessed by unauthorized users, it can lead to data exposure 
+   and potential security breaches.You can mitigate this through:
+
+   - Restricting access: Ensure that the checkpoint archive is accessible only
+     by authorized users. Set appropriate file permissions and access controls
+     to limit access to the archive.
+   
+   - Encryption: Encrypt the checkpoint archive to protect the data stored 
```

Comment:
There is a PR open to encrypt the checkpoint before being written to disk: https://github.com/checkpoint-restore/criu/pull/2297

This will happen on the CRIU level and the data will never hit storage unencrypted. If it is merged in CRIU we need to add it to runc/crun and CRI-O/containerd. The support on the layers above CRIU will basically be calling the layer below with the right options. All the actual encryption work will happen in CRIU.